runtime/yaml.cpp

```diff
@@ -0,0 +1,191 @@
+  } else if (node.IsSequence()) {
+    for (auto it = node.begin(); it != node.end(); ++it) {
+      mixed data_piece;
+      yaml_node_to_mixed(*it, data_piece, source);
+      data.push_back(data_piece);
+    }
+  } else if (node.IsMap()) {
+    for (const auto &it : node) {
+      mixed data_piece;
+      yaml_node_to_mixed(it.second, data_piece, source);
+      data.set_value(string(it.first.as<std::string>().c_str()), data_piece);
+    }
+  }
+  // else node is Null or Undefined, so data is Null
+}
+
+/*
+ * print tabs in quantity of nesting_level (used to print nested YAML entries)
+ */
+static string yaml_print_tabs(const uint8_t nesting_level) noexcept {
+  return string(2 * nesting_level, ' ');
+}
+
+/*
+ * print the key of a YAML map entry
+ */
+static string yaml_print_key(const mixed &data_key) noexcept {
+  if (data_key.is_string()) {
+    return data_key.as_string();
+  }
+  return string(data_key.as_int()); // key can not be an array; bool and float keys are cast to int
+}
+
+/*
+ * escape special characters in a string entry
+ */
+static string yaml_escape(const string &data) noexcept {
+  string escaped_data;
+  for (size_t i = 0; i < data.size(); ++i) {
+    if (data[i] == 10) { // line feed - code 10
+      escaped_data.push_back(92); // backslash
+      escaped_data.push_back('n');
+    } else if (data[i] == 8) { // backspace - code 8
+      escaped_data.push_back(92); // backslash
+      escaped_data.push_back('b');
+    } else if (data[i] == 9) { // horizontal tab - code 9
+      escaped_data.push_back(92); // backslash
+      escaped_data.push_back('t');
+    } else if (data[i] == 11) { // vertical tab - code 11
+      escaped_data.push_back(92); // backslash
+      escaped_data.push_back('v');
+    } else if (data[i] == 34) { // double quotation mark - code 34
+      escaped_data.push_back(92); // backslash
+      escaped_data.push_back(34); // double quotation mark
+    } else if (data[i] == 92) { // backslash - code 92
+      escaped_data.push_back(92);
+      escaped_data.push_back(92); // double backslash
+    } else {
+      escaped_data.push_back(data[i]);
+    }
+  }
+  return escaped_data;
+}
+
+/*
+ * get a YAML representation of mixed in a string variable
+ */
+static void mixed_to_string(const mixed &data, string &string_data, const uint8_t nesting_level = 0) noexcept {
+  string buffer;
+  if (!data.is_array()) {
+    if (data.is_null()) {
+      buffer.push_back('~'); // tilda is a YAML representation of NULL
+    } else if (data.is_string()) {
+      buffer.push_back('"'); // cover string entry in double quotes
+      buffer.append(yaml_escape(data.as_string())); // escape special characters
+      buffer.push_back('"');
+    } else if (data.is_int()) {
+      buffer.append(data.as_int());
+    } else if (data.is_float()) {
+      buffer.append(data.as_double());
+    } else if (data.is_bool()) {
+      buffer = (data.as_bool()) ? string("true") : string("false");
+    }
+    string_data.append(buffer);
```

Comment:
What for do you need `buffer`, why not to append into `string_data` directly?

Reply:
`buffer` was added in prevoius iteration of review, I think it makes code more readable, for example on line 118